src/Spryker/Client/RabbitMq/RabbitMqFactory.php

```diff
@@ -151,4 +158,24 @@ public function createQueueEstablishmentHelper(): QueueEstablishmentHelperInterf
     {
         return new QueueEstablishmentHelper();
     }
+
+    /**
+     * @return \Spryker\Client\RabbitMq\Model\Queue\QueueMetricReaderInterface
+     */
+    public function createQueueMetricReader(): QueueMetricReaderInterface
+    {
+        return new QueueMetricReader($this->getStaticConnectionManager());
+    }
+
+    /**
+     * @return \Spryker\Client\RabbitMq\Model\Queue\QueueMetricReaderInterface
+     */
+    public function getQueueMetricReader(): QueueMetricReaderInterface
+    {
+        if (static::$queueMetricReader === null) {
+            static::$queueMetricReader = $this->createQueueMetricReader();
+        }
```

Comment:
Any reason to use the same object? (I don't see any cache inside)

Reply:
To avoid many creations of this object